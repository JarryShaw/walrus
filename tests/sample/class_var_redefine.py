import os


class A:
    print('Redefine with variable')
    print(var := 21)
    var = 61
    print(var)
    print(vars()['var'])

    print('Redefine with function')
    print(func := 22)

    @staticmethod
    def func(var):
        print('foo called')
        return var

    print(type(func))

    print('Redefine with class')
    print(cls := 23)

    class cls:
        pass

    print(type(cls))

    print('Redefine with for Loop')
    print(loopvar := 24)

    for loopvar in range(64, 65):
        print(loopvar)

    [print(loopvar) for loopvar in range(74, 75)]  # loopvar should be local to this comprehension
    print(loopvar)

    print('Redefine with import')
    print(mod := 25)
    import sys as mod
    print(type(mod))
    print(item := 26)
    from sys import version as item
    print(isinstance(item, int))

    print('Redefine with with statement')
    print(ctx := 27)
    with open(os.devnull) as ctx:
        pass
    print(isinstance(ctx, int))

    print('Redefine with except statement')
    print(exc := 28)
    try:
        raise RuntimeError
    except RuntimeError as exc:
        print(type(exc))
    print(locals().get('exc'))

    print('Class A initialized')


print('var = {}'.format(A.var))
print('func is of type {}'.format(type(A.func)))
print('func execution returns: {}'.format(A.func(62)))
print('cls is of type: {}'.format(type(A.cls)))
print('loopvar = {}'.format(A.loopvar))
print('mod is of type {}'.format(type(A.mod)))
print('item is int? {}'.format(isinstance(A.item, int)))
print('ctx is int? {}'.format(isinstance(A.ctx, int)))
print('exc in A: {}'.format(A.__dict__.get('exc')))
