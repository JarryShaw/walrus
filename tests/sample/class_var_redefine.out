Redefine with variable
21
61
61
Redefine with function
22
<class 'staticmethod'>
Redefine with class
23
<class 'type'>
Redefine with for Loop
24
64
74
64
Redefine with import
25
<class 'module'>
26
False
Redefine with with statement
27
False
Redefine with except statement
28
<class 'RuntimeError'>
None
Class A initialized
var = 61
func is of type <class 'function'>
foo called
func execution returns: 62
cls is of type: <class 'type'>
loopvar = 64
mod is of type <class 'module'>
item is int? False
ctx is int? False
exc in A: None
